metric-schema-java/src/integrationInput/java/com/palantir/test/NamespaceTagsMetrics.java

```diff
@@ -0,0 +1,196 @@
+                .putSafeTags("noValueTag", noValueTag)
+                .putSafeTags("locatorWithMultipleValues", locatorWithMultipleValues.getValue())
+                .putSafeTags("otherLocator2", "package:identifier")
+                .putSafeTags("libraryName", LIBRARY_NAME)
+                .putSafeTags("libraryVersion", LIBRARY_VERSION)
+                .putSafeTags("javaVersion", JAVA_VERSION)
+                .build());
+    }
+
+    @Override
+    public String toString() {
+        return "NamespaceTagsMetrics{registry=" + registry + '}';
+    }
+
+    public enum NamespaceTags_LocatorWithMultipleValues {
+        PACKAGE_IDENTIFIER("package:identifier"),
+
+        PACKAGE_IDENTIFIER2("package:identifier2");
+
+        private final String value;
+
+        NamespaceTags_LocatorWithMultipleValues(String value) {
+            this.value = value;
+        }
+
+        private String getValue() {
+            return value;
+        }
+    }
+
+    public enum Processing_Result {
+        SUCCESS("success"),
+
+        FAILURE("failure");
+
+        private final String value;
+
+        Processing_Result(String value) {
+            this.value = value;
+        }
+
+        private String getValue() {
+            return value;
+        }
+    }
+
+    public enum Processing_OtherLocator {
+        PACKAGE_IDENTIFIER("package:identifier"),
+
+        PACKAGE_IDENTIFIER2("package:identifier2");
+
+        private final String value;
+
+        Processing_OtherLocator(String value) {
+            this.value = value;
+        }
+
+        private String getValue() {
+            return value;
+        }
+    }
+
+    public interface ProcessingBuildStage {
+        @CheckReturnValue
+        Meter build();
+    }
+
+    public interface ProcessingBuilderResultStage {
+        /**
+         * The result of processing
+         */
+        @CheckReturnValue
+        ProcessingBuilderTypeStage result(@Safe Processing_Result result);
+    }
+
+    public interface ProcessingBuilderTypeStage {
+        @CheckReturnValue
+        ProcessingBuilderOtherLocatorStage type(@Safe String type);
+    }
+
+    public interface ProcessingBuilderOtherLocatorStage {
+        @CheckReturnValue
+        ProcessingBuildStage otherLocator(@Safe Processing_OtherLocator otherLocator);
+    }
+
+    private final class ProcessingBuilder
+            implements ProcessingBuilderResultStage,
+                    ProcessingBuilderTypeStage,
+                    ProcessingBuilderOtherLocatorStage,
+                    ProcessingBuildStage {
+        private Processing_Result result;
+
+        private String type;
+
+        private Processing_OtherLocator otherLocator;
+
+        @Override
+        public Meter build() {
+            return registry.meter(MetricName.builder()
+                    .safeName("namespace-tags.processing")
+                    .putSafeTags("locator", "package:identifier")
+                    .putSafeTags("noValueTag", noValueTag)
```

Comment:
This fallthrough to outer class is very serendipidous, I'm not sure we've made the inner classes non-static on purpose. I think I'd like to ship with this and maybe refactor at some other time; I guess we're trading memory locality vs memory usage?

Reply:
I think it was intentional as a way to access the `registry`. The namespaced tag access is no different, I like it

Reply:
ah, missed that! perfect then